scene/src/utils/wearables-promise-utils.ts

```diff
@@ -0,0 +1,118 @@
+import type {
+  CatalogWearableElement,
+  WearableEntityMetadata,
+  CatalystWearableMap
+} from './wearables-definitions'
+import type { URNWithoutTokenId } from './definitions'
+import { type WearableCategory } from '../service/wearable-categories'
+import { getRealm } from '~system/Runtime'
+
+export type WearablesPageResponse = {
+  elements: CatalogWearableElement[]
+  pageNum: number
+  pageSize: number
+  totalAmount: number
+}
+
+export type WearableCatalogPageParams = {
+  pageNum: number
+  pageSize: number
+  address: string
+  wearableCategory: WearableCategory | null
+  includeBase: boolean
+  includeOnChain: boolean
+  catalystBaseUrl: string
+}
+
+// cache
+export const catalystWearableMap: CatalystWearableMap = {}
+
+export async function fetchWearablesPage({
+  pageNum,
+  pageSize,
+  wearableCategory,
+  address
+}: any): Promise<WearablesPageResponse> {
```

Comment:
```suggestion
}: WearableCatalogPageParams): Promise<WearablesPageResponse> {
```

can we try to avoid the `any` here?

Reply:
https://github.com/dclexplorer/bevy-ui-scene/commit/cae232fb9cd7ddabbe91374a0f23f5d5e55469a1#diff-44430281a4df0b2fc3bfb512bd4c191a352664a9c4b4ae8c1abf0c11499fe73d